cpp.md

```diff
@@ -0,0 +1,163 @@
+# C++ conventions
+
+### Compilation
+
+The table below contains the specifics of environment to build the projects. These settings are recommended and please make sure your project and machine matches these specs.
+
+| Topic                    | Config                                              |
+|--------------------------|-----------------------------------------------------|
+| C++ version              | C++ 14                                              |
+| Compiler                 | GCC / CLANG (11.0 +)                                |
+| Platforms                | iOS, android, amazon                                |
+| Build tools              | CMake (3.6+), gradle for android (5.x), pbx for iOS |
+| Compiler flags           | Warnings and Errors                                 |
+| Optional libs (external) | Boost                                               |
+
+We are also using CPP check to avoid any false positives in the project. You can easily install it on your system and learn more about it from - http://cppcheck.sourceforge.net/
+
+### Coding Style
+
+**Class name -** use camel case with first upper. eg
+``` C++
+class Person
+{
+  // ..
+};
+```
+
+**local variable name -**  use snake type. eg
+``` C++
+std::string first_name = “sameer”
+```
+
+**local class variables -** should be camel case with first letter lower case, eg
+``` C++
+SampleClass sampleObj;
+```
+
+**member class variables -** these are member class variables (objects), should be camel case with first letter lower case  and prefixed with m_.  public class variables should be named as local class variables
+``` C++
+SampleClass *m_sampleObj;
+```
+
+**function names -** use camelcase with first letter lower, eg:
+``` C++
+void doSomethingForMe ()
+{
+  // ..
+}
+```
+
+**function params -** optionally, you can precede params with p_ when params in function scope named similar to params to avoid confusion. params follow same naming convention as local variables. 
```

Comment:
no optionally please. Let's be decisive and go with that

Reply:
I kept it so people can know such sol is there. while reading large functions which deal with lots of variables (in our case when vyshnav was trying to call func to main thread) it becomes hard to know where the variable is coming from and this saves some brain cells while reading code and tracking them down.

However, I'll remove this.

Reply:
Understand, but will be difficult for the pre-commit hook to check this.